legal-api/src/legal_api/services/search_service.py

```diff
@@ -0,0 +1,268 @@
+
+    # Function to check if codes belong in BUSINESS_TEMP_FILINGS_CORP_CODES
+    @staticmethod
+    def check_and_get_respective_values(codes):
+        result = {}
+        
+        for code in codes:
+            if code in BusinessSearchService.BUSINESS_TEMP_FILINGS_CORP_CODES:
+                result[code] = BusinessSearchService.BUSINESS_TEMP_FILINGS_CORP_CODES[code]
+            else:
+                result[code] = None
+        
+        return result
+
+    @classmethod
+    def separate_states_by_type(cls, states: List[str]) -> Tuple[List[str], List[str]]:
+        """
+        Separates input states into business-eligible and filing-eligible states,
+        using human-readable names (e.g. 'Active') and mapping them to enum values.
+        """
+        input_normalized = [s.upper() for s in states]
+
+        business_name_to_value = {state.name: state.value for state in Business.State}
+        filing_name_to_value = {status.name: status.value for status in Filing.Status}
+
+        business_states = [
+            business_name_to_value[state]
+            for state in input_normalized
+            if state in business_name_to_value and business_name_to_value[state] in cls.BUSINESS_ELIGIBLE_STATES
+        ]
+
+        filing_states = [
+            filing_name_to_value[state]
+            for state in input_normalized
+            if state in filing_name_to_value and filing_name_to_value[state] in cls.FILINGS_ELIGIBLE_STATES
+        ]
+        return business_states, filing_states
+
+
+    @staticmethod
+    def try_parse_legal_type(legal_type: str) -> bool:
+        try:
+            Business.LegalTypes(legal_type)
+            return True
+        except ValueError:
+            return False
+        
+    @staticmethod
+    def separate_legal_types(legal_types: List[str]) -> Tuple[List[str], List[str]]:
+        """
+        Separates input legal types into valid and invalid based on the Business.LegalTypes enum.
+        """
+        input_normalized = [t.strip().upper() for t in legal_types]
+
+        valid_types = [t for t in input_normalized if BusinessSearchService.try_parse_legal_type(t)]
+        invalid_types = [t for t in input_normalized if not BusinessSearchService.try_parse_legal_type(t)]
+        return valid_types, invalid_types
+    
+    @staticmethod
+    def get_search_filtered_businesses_results(business_json, identifiers=None, search_filters: AffiliationSearchDetails = None):
+        """Return contact point from business json."""
+
+        def _get(attr, default):
+            return getattr(search_filters, attr, default) if search_filters else default
+ 
+        name = _get('name', None)
+        types = _get('type', [])
+        statuses = _get('status', [])
+        identifier = _get('identifier', None)
+        
+        valid_types, _ = BusinessSearchService.separate_legal_types(types or [])
+
+        #edge case: if type searched for doesnt belong to table business such as 'ATMP' Returns early
```

Comment:
Good comment, spacing at the front.. shows intent that isn't obvious by reading the code